### ISSUE TITLE
RetryPolicy object provided through WithPolicy setter is overwintered with default RetryPolicy.

### ISSUE THREAD

**Issue by hovsepm** [Azure/azure-libraries-for-net#479]
Here are some more code provided from Customer that they implement the retry policy and setup 
```
RestClient and RetryPolicy
 
namespace Zerto.Zvm.Platform.PublicCloud.AzureProxy.Implementation
{
    public class FluentWrapperFailureStrategy : ITransientErrorDetectionStrategy
    {
        public bool IsTransient(Exception ex)
        {
              return true;
        }
    }
}
``` 
``` 
var retryPolicy = new RetryPolicy<FluentWrapperFailureStrategy>(retryCount, TimeSpan.FromSeconds(retryInterval));
var servicePrincipalLogin = new ServicePrincipalLoginInformation
{
    ClientId = appCreds.AppId,
    ClientSecret = appCreds.Secret
};

var azureEnvironment = AzureBaseUriMapper.GetAzureEnvironment(location);
AzureCredentials azureCredentials =
    new AzureCredentials(servicePrincipalLogin, tenantId, azureEnvironment);
IAzureFluentWrapper azure =
    AzureFluentWrapper.Configure()
                      .WithRetryPolicy(retryPolicy)
                      .WithLogLevel(c_httpLoggingDelegatingHandlerLevel)
                      .WithDelegatingHandler(new HttpLoggingTracingHandler())
                      .Authenticate(azureCredentials)
                      .WithSubscription(subscription);
```

Customer Request:
 How do they implement retry logic when using Azure Management Library?

**Comment by Digmaster** [comment]
Is there a viable workaround for this issue in the meantime?

**Comment by hovsepm** [comment]
@Digmaster unfortunately no. This is a bug that involves shared ClientRuntime component as well. Fix is already done and in the testing phase. Will be available in release 1.16.1.

**Comment by hovsepm** [comment]
Release 1.16.1 is out. Please upgrade your references. Please open a new issue if you will hit same/new errors.
